proxy/src/config.rs

```diff
@@ -340,6 +339,42 @@ fn parse_number<T>(s: &str) -> Result<T, ParseError> where T: FromStr {
     s.parse().map_err(|_| ParseError::NotANumber)
 }
 
+fn parse_duration(s: &str) -> Result<Duration, ParseError> {
+    use regex::Regex;
+    lazy_static! {
+        static ref RE: Regex = Regex::new(r"(?x)^
+            \s* (\d+) \s*
+            ( ms | msecs? | millis?  | milliseconds?
+            | s  | secs?  | seconds?
+            | m  | mins?  | minutes?
+            | h  | hours?
+            | d  | days?
+            ) \s*
+        $").expect("duration regex");
+    }
+
+    let cap = RE.captures(s)
+        .ok_or(ParseError::NotADuration)?;
+
+    let magnitude = parse_number(&cap[1])?;
+
+    let unit = &cap[2];
+    debug_assert!(unit.len() >= 1);
+
+    if unit.starts_with("ms") || unit.starts_with("mil") {
+        return Ok(Duration::from_millis(magnitude));
+    }
+
+    let to_secs = match unit.chars().next() {
+        Some('s') => 1,
+        Some('m') => 60,
+        Some('h') => 60 * 60,
+        Some('d') => 60 * 60 * 24,
+        _ => return Err(ParseError::NotADuration),
+    };
```

Comment:
In particular, notice that this doesn't validate anything after the first character, which I'm sure will create problems later, if we change anything here later.

Instead, we should match `unit` as full strings. Then we can also merge the processing for "ms" into the match and use `from_millis` for everything.